### PR TITLE
{data}[foss/2018b] HDF5 v1.8.12

### PR DESCRIPTION
**Pull request by sassy-crick**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by akesandgren** [comment]
Any specific reason you need this old version? We have HDF5 1.10.12 for 2018b toolchains already.

**Comment by sassy-crick** [comment]
@akesandgren Yes. I got a proprietary RPM which requires this version as else you get an error message:
Headers are 1.8.12, library is 1.8.13
I originally used 1.8.13 which threw the error message and thus I downgraded to 1.8.12. So for that reason, as others might run into the same problem, I thought it would be nice to share the build file. 
I leave it up to you guys to decide the fate of this and the PR # 8290. Thanks

File: easybuild/easyconfigs/h/HDF5/HDF5-1.8.12-foss-2018b.eb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    'b5cccea850096962b5fd9e96f22c4f47d2379224bb41130d9bc038bb6c37dfcb',  # HDF5-tarball
+    '44e04cace87c54f75e8eb5b710679104f6703d88d10d6f3e5130b8cd90c69aa1',  # configure_libtool.patch
+]
+
+patches = [
+    'configure_libtool.patch',
+]
+
+dependencies = [
+    ('zlib', '1.2.8'),
```
Comment (akesandgren) [inline easybuild/easyconfigs/h/HDF5/HDF5-1.8.12-foss-2018b.eb]: For 2018b toolchains use 1.2.11 instead, that is already available at GCCcore level.

Comment (sassy-crick) [reply easybuild/easyconfigs/h/HDF5/HDF5-1.8.12-foss-2018b.eb]: I thought of that but I was not sure if that might break things during the compilation of HDF5. I will give it a go as Travis has failed again.

**Comment by akesandgren** [comment]
Please drop the zlib easyconfig from this PR too.

eb --update-pr=8289 :zlib-1.2.8-GCCcore-7.3.0.eb --pr-commit-msg='Drop not used zlib easyconfig'

Or something like that in the commit msg

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3150.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/1c24be410fc6c72d8cf9d94bff868107 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2454.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/2fa3d772077739f7af7e422e65ba3455 for a full test report.

File: easybuild/easyconfigs/h/HDF5/HDF5-1.8.12-foss-2018b.eb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+name = 'HDF5'
+version = '1.8.12'
+
+homepage = 'https://support.hdfgroup.org/HDF5/'
+description = """HDF5 is a data model, library, and file format for storing and managing data.
+ It supports an unlimited variety of datatypes, and is designed for flexible
+ and efficient I/O and for high volume and complex data."""
+
+toolchain = {'name': 'foss', 'version': '2018b'}
+toolchainopts = {'optarch': True, 'pic': True, 'usempi': True}
```
Comment (boegel) [inline easybuild/easyconfigs/h/HDF5/HDF5-1.8.12-foss-2018b.eb]: @sassy-crick Please drop the `'optarch': True, ` (since that's enabled by default)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2120.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/abbaf6113544af496bb8944448a7c798 for a full test report.

File: easybuild/easyconfigs/h/HDF5/HDF5-1.8.12-foss-2018b.eb (hunk 2 above)
Comment (sassy-crick) [reply easybuild/easyconfigs/h/HDF5/HDF5-1.8.12-foss-2018b.eb]: Done. However, it should be noted that it still was in the HDF5-1.8.13-foss-2018b.eb which already made it into the repository.

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
nic425 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/61380075aa9904eee133a1597da24cc9 for a full test report.

**Comment by smoors** [comment]
Going in, thanks @sassy-crick!

**Comment by smoors** [comment]
Going in, thanks @sassy-crick!

**Comment by smoors** [comment]
Going in, thanks @sassy-crick!

**Pull request merged**